src/bindable-property.js

```diff
@@ -35,7 +35,13 @@ export class BindableProperty {
   * Creates an instance of BindableProperty.
   * @param nameOrConfig The name of the property or a cofiguration object.
   */
-  constructor(nameOrConfig: string | Object) {
+  constructor(nameOrConfig: string | {
```

Comment:
Thanks for typing that `Object`!
Should we have a named interface for that rather? `BindableOptions`? That would make the code cleaner + give a place to comment on members... I know that `attribute` does but it's not totally obvious, especially now with `reflect` in the mix.